node/heartbeat/hearbeatMessageInfo.go

```diff
@@ -37,26 +40,22 @@ func (hbmi *heartbeatMessageInfo) clockTime() time.Time {
 func (hbmi *heartbeatMessageInfo) sweep() {
 	crtDuration := hbmi.getTimeHandler().Sub(hbmi.timeStamp)
 	hbmi.isActive = crtDuration < hbmi.maxDurationPeerUnresponsive
-	if hbmi.maxInactiveTime.Duration < crtDuration {
-		hbmi.maxInactiveTime.Duration = crtDuration
-	}
+	hbmi.updateMaxInactiveTimeDuration()
 }
 
-// HeartbeatReceived processes a new message arrived from a p2p address
+// HeartbeatReceived processes a new message arrived from a peer
 func (hbmi *heartbeatMessageInfo) HeartbeatReceived(shardID uint32) {
 	crtTime := hbmi.getTimeHandler()
 	hbmi.sweep()
+	hbmi.isActive = true
+	hbmi.shardID = shardID
+	hbmi.updateMaxInactiveTimeDuration()
+	hbmi.timeStamp = crtTime
+}
 
-	if !hbmi.alreadyAccessed {
-		hbmi.shardID = shardID
-		hbmi.isActive = true
-		hbmi.timeStamp = crtTime
-		hbmi.maxInactiveTime = Duration{Duration: 0}
-	}
-	crtDuration := crtTime.Sub(hbmi.timeStamp)
-	if hbmi.maxInactiveTime.Duration < crtDuration {
+func (hbmi *heartbeatMessageInfo) updateMaxInactiveTimeDuration() {
+	crtDuration := hbmi.getTimeHandler().Sub(hbmi.timeStamp)
+	if hbmi.maxInactiveTime.Duration < crtDuration && hbmi.timeStamp != (time.Time{}) {
```

Comment:
you should set maxInactive to maxDuration, han you do not need the second part of the condition

Reply:
not possible because maxInactive will be compared to the difference between current time and timestamp and the 'if' clause. The default value for maxDurationPeerUnresponsive is 1 min. Also, providing a maximum duration value would not be ok. 
However, I added a variable called emptyTimestamp so it would be more readable. Also, the second condition of the _if_ won't be calculated very often, because most of the time the first condition won't pass